liblocation/src/main/java/com/mapbox/android/core/location/GoogleLocationEngine.java

```diff
@@ -147,12 +165,19 @@ public Type obtainType() {
     return Type.GOOGLE_PLAY_SERVICES;
   }
 
-  @Override
-  public void onLocationChanged(Location location) {
-    for (LocationEngineListener listener : locationListeners) {
-      listener.onLocationChanged(location);
+  private LocationCallback locationCallback = new LocationCallback() {
+    @Override
+    public void onLocationResult(LocationResult locationResult) {
+      List<Location> locationList = locationResult.getLocations();
+      if (locationList.size() > 0) {
```

Comment:
nit use `!list.isEmpty()` over `locationList.size() > 0`, depending on the List implementation this can result in  overhead.